clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractMembershipManager.java

```diff
@@ -189,6 +191,9 @@ public abstract class AbstractMembershipManager<R extends AbstractResponse> impl
 
     private final Time time;
 
+    // AtomicBoolean to track if the subscription has been updated
```

Comment:
this is truly more to know if we should join after the subscription has been updated right? (true if not in group)